src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -51,4 +54,25 @@ private[spark] object Config {
       .timeConf(TimeUnit.SECONDS)
       .checkValue(interval => interval > 0, s"Timeout must be a positive time value.")
       .createWithDefaultString("5")
+
+  // See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+  // Clients do not use the prefix, therefore we just accept the name portion on label selector names.
+  private val label = "([a-z0-9A-z&&[^-_.]]([a-z0-9A-Z-_.]{0,61}[a-z0-9A-Z&&[^-_.]])?)"
+  private val labelSelectors: Regex = (s"^($label=$label(,$label=$label)*)?$$").r
+
+  private[armada] val selectorsValidator = selectors => labelSelectors.matches(selectors)
```

Comment:
I just noticed the CI compilations are failing, due to this line. Does `selectorsValidator` need a type, something like `List[String]` ?

Reply:
I think I fixed this in 1d0c7ac